OutOfSchool/OutOfSchool.AuthCommon/Services/ProviderAdminService.cs

```diff
@@ -50,6 +51,7 @@ public class ProviderAdminService : IProviderAdminService
         this.gPRCConfig = gRPCConfig?.Value ?? throw new ArgumentNullException(nameof(gRPCConfig));
         this.externalUrisConfig =
             externalUrisConfig?.Value ?? throw new ArgumentNullException(nameof(externalUrisConfig));
+        this.changesLogConfig = changesLogConfig?.Value ?? throw new ArgumentNullException(nameof(changesLogConfig));
     }
 
     public async Task<ResponseDto> CreateProviderAdminAsync(
```

Comment:
(Code style, but I understand that it's not yours, so not insisting on fix) 
`CreateProviderAdminAsync` method is "smelly code". It spans over lines 57-210 and as for me is too long. Also it contains nested lambda, where main logic resides, Also there is repeated pattern for setting `ResponseDto` fields (that might be separated as method of that class). Probably there are more changes to approve readability.
Please consider it as an example of not good code, and (if you brave enaugh) rewrite it, or (at least) avoid adding more code in such methods. 
For now, I'm not insisting on changes, because it is out of this PR scope

Reply:
Hello Vadym, extracted the comments to separate [issue](https://github.com/ita-social-projects/OoS-Backend/issues/1258) so we can deal with it in new PR.

Reply:
Thanks @DmyMi!